### PR TITLE
Added jquery for drag and drop of pieces.

### PR DESCRIPTION
**Pull request by tbowyer**
Added a pieces controller and an update function to handle updating the moving pieces attributes.

Modified the routes.rb file to include a route for each individual piece that will submit a PATCH request.  This is for the AJAX after dropping a piece.

Removed the code from the _draw_pieces partial to remove all the links and added data-ids to the square divs and pieces.

Added jquery code in the main games/show view for the drag and drop.

### CODE REVIEW

File: app/assets/javascripts/pieces.js.coffee [hunk 1]
```diff
@@ -0,0 +1,3 @@
+# Place all the behaviors and hooks related to the matching controller here.
```
Comment (ilyakrasnov) [inline app/assets/javascripts/pieces.js.coffee:1]: Do we use coffescript? If not, you can delete this file and leave pieces.js if needed

File: app/controllers/pieces_controller.rb [hunk 2]
```diff
@@ -0,0 +1,22 @@
+class PiecesController < ApplicationController
```
Comment (ilyakrasnov) [inline app/controllers/pieces_controller.rb:1]: :+1:

File: app/models/pieces/bishop.rb [hunk 3]
```diff
@@ -5,4 +5,8 @@ def set_defaults
     self.moved = false
     self.icon = (self.color == 1) ? "&#9821;" : "&#9815;"
   end
+
+  def valid_move?(x,y)
+  	return false unless ((x_position - x) / (y_position - y)) == 1 or -1
```
Comment (ilyakrasnov) [inline app/models/pieces/bishop.rb:10]: Maybe a good use for a private method as discussed in the call:

return false unless valid_diagonal_move?(x,y) and putting the slope logic there

**Comment by travisdilweg** [comment]
Any reasons we haven't merged this yet?

**Pull request merged**